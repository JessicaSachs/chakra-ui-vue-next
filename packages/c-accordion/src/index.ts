import { h, defineComponent, PropType } from 'vue'
import { chakra, DOMElements } from '@chakra-ui/vue-system'

const CAccordion = defineComponent({
  props: {
    as: {
      type: [String] as PropType<DOMElements>,
      default: 'div',
    },
  },
  setup(props, { slots, attrs }) {
    return () => h(chakra(props.as), { ...attrs }, slots)
  },
})

export default CAccordion
