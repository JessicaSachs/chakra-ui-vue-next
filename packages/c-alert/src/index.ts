import { h, defineComponent, Component } from 'vue'

const CAlert = defineComponent({
  props: {
    as: {
      type: [String, Object],
      default: 'div',
    },
  },
  render() {
    return h(
      this.as,
      {
        ...this.$attrs,
        role: 'alert',
      },
      this.$slots.default && this.$slots.default()
    )
  },
})

export default CAlert
