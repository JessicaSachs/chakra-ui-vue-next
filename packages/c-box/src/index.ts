import { h, defineComponent, PropType } from 'vue'

const CBox = defineComponent({
  props: {
    as: {
      type: String as PropType<string>,
      default: 'div',
    },
  },
  render() {
    return h(this?.as, { ...this.$props, ...this.$attrs }, this.$slots)
  },
})

export default CBox
