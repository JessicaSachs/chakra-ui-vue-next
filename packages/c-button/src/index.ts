import { h, defineComponent, PropType } from 'vue'

const CButton = defineComponent({
  name: 'CButton',
  props: {
    as: {
      type: String as PropType<string>,
      default: 'button',
    },
  },
  render() {
    return h(
      this?.as,
      { ...this.$props, ...this.$attrs },
      this.$slots.default && this.$slots.default()
    )
  },
})

export default CButton
