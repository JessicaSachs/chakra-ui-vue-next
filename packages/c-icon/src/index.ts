export * from './icon'
export { default as internalIcons } from './icon.internals'
export * from './icon.internals'
