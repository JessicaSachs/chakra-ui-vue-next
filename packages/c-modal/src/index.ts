export { default as CModal } from './c-modal'
