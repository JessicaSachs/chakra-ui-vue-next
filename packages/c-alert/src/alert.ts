import { h, defineComponent, PropType, provide, inject, computed } from 'vue'
import {
  chakra,
  ColorScheme,
  DeepComponentThemeConfig,
  ThemingProps,
  useMultiStyleConfig,
  provideComponentStyles,
  useComponentStyles,
  DOMElements,
} from '@chakra-ui/vue-system'
import { SystemStyleObject } from '@chakra-ui/styled-system'
import { CIcon } from '@chakra-ui/c-icon'

const STATUSES = {
  info: {
    colorScheme: 'blue' as ColorScheme,
    icon: 'info',
  },
  success: {
    colorScheme: 'green' as ColorScheme,
    icon: 'check-circle',
  },
  warning: {
    colorScheme: 'orange' as ColorScheme,
    icon: 'warning-alt',
  },
  error: {
    colorScheme: 'red' as ColorScheme,
    icon: 'warning',
  },
}

type AlertStatus = keyof typeof STATUSES
export type AlertVariant = 'solid' | 'subtle' | 'left-accent' | 'top-accent'

interface AlertState {
  status: AlertStatus
}

/**
 * CAlert component
 *
 * This is the container component for all Alert components.
 * It also provides state and context to it's compound components
 */
export const CAlert = defineComponent({
  name: 'CAlert',
  props: {
    as: {
      type: [String, Object] as PropType<DOMElements>,
      default: 'div',
    },
    status: {
      type: [String] as PropType<AlertStatus>,
      default: 'info',
    },
    colorScheme: {
      type: [String] as PropType<ColorScheme>,
    },
    styleConfig: {
      type: [Object] as PropType<DeepComponentThemeConfig>,
    },
    variant: {
      type: [String] as PropType<AlertVariant>,
      default: 'solid',
    },
  },
  setup(props, { slots, attrs }) {
    const colorScheme: ColorScheme =
      props.colorScheme || STATUSES[props.status].colorScheme

    const themingProps: ThemingProps = {
      colorScheme,
      variant: props.variant,
    }
    const styles = useMultiStyleConfig('Alert', themingProps)
    const alertStyles: SystemStyleObject = {
      width: '100%',
      display: 'flex',
      alignItems: 'center',
      position: 'relative',
      overflow: 'hidden',
      ...styles.value.container,
    }

    provideComponentStyles('Alert', styles.value)
    provide('$AlertState', { status: props.status } as AlertState)

    return () =>
      h(
        chakra(props.as, 'alert'),
        {
          role: 'alert',
          ...alertStyles,
          ...attrs,
        },
        slots
      )
  },
})

/**
 * CAlertTitle component
 *
 * The title component for alerts
 */
export const CAlertTitle = defineComponent({
  name: 'CAlertTitle',
  setup(_, { attrs, slots }) {
    const styles = useComponentStyles('Alert')

    return () =>
      h(
        chakra('div', 'alert__title'),
        {
          ...styles.title,
          ...attrs,
        },
        slots
      )
  },
})

/**
 * CAlertDescription component
 *
 * The description component for alerts
 */
export const CAlertDescription = defineComponent({
  name: 'CAlertDescription',
  setup(_, { attrs, slots }) {
    const styles = useComponentStyles('Alert')

    return () =>
      h(
        chakra('div', 'alert__description'),
        {
          ...styles.description,
          ...attrs,
        },
        slots
      )
  },
})

/**
 * CAlertIcon component
 *
 * The Icon component for alerts
 */
export const CAlertIcon = defineComponent({
  name: 'CAlertIcon',
  props: {
    icon: {
      type: [String] as PropType<string>,
    },
  },
  setup(props, { attrs }) {
    const alertState = inject<AlertState>('$AlertState')
    const { icon } = STATUSES[alertState?.status as AlertStatus]
    const styles = useComponentStyles('Alert')

    const alertIcon = computed(() => props.icon || icon)

    return () =>
      h(CIcon, {
        class: 'alert__icon',
        name: alertIcon.value,
        ...styles.icon,
        ...attrs,
      })
  },
})
