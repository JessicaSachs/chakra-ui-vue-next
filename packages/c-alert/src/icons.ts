import { defineComponent, h } from 'vue'
import { CIcon } from '@chakra-ui/c-icon'

export const CCheckIcon = defineComponent({
  name: 'CCheckIcon',
  inheritAttrs: false,
  setup(_, { attrs }) {
    return () =>
      h(CIcon, {
        name: 'check-circle',
        attrs,
      })
  },
})
