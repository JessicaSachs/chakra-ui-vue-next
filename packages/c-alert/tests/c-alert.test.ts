import { render } from '@chakra-ui/vue-test-utils'
import { CAlert } from '../src'

describe('Alert tests', () => {
  it('should render properly', () => {
    const { asFragment } = render(CAlert)
    expect(asFragment()).toMatchSnapshot()
  })
})
