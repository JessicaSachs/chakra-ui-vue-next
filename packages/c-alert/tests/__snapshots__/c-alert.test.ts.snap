// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render all children 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-wtzaz1"
    data-testid="alert"
    role="alert"
  >
    <svg
      class="alert__icon chakra-icon css-139yi2r"
      focusable="false"
      viewBox="0 0 24 24"
    >
      
    
      <g
        stroke="currentColor"
        strokewidth="1.5"
      >
        
      
        <path
          d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
          fill="none"
          strokelinecap="round"
        />
        
      
        <path
          d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
          fill="currentColor"
          strokelinecap="round"
        />
        
      
        <circle
          cx="12"
          cy="12"
          fill="none"
          r="11.25"
          strokemiterlimit="10"
        />
        
    
      </g>
      
  
    </svg>
    <div
      class="chakra-alert__title css-14esyki"
    >
       Info alert 
    </div>
    <div
      class="chakra-alert__description css-5le3e4"
    >
       Something just happened 
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-15689h3"
    role="alert"
  />
</DocumentFragment>
`;
